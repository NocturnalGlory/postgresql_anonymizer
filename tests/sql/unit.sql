
CREATE EXTENSION IF NOT EXISTS anon;

--                                                                                                                                                     
-- Generic Types                                                                                                                                       
--     

-- zip
SELECT pg_typeof(anon.random_zip()) = 'TEXT'::REGTYPE;


-- string

SELECT pg_typeof(anon.random_string(1)) = 'TEXT'::REGTYPE;
--SELECT anon_string(123456789);


-- Date                                                                                                                                                
SELECT pg_typeof(anon.random_date_between('01/01/1900'::TIMESTAMP WITH TIME ZONE,now())) = 'TIMESTAMP WITH TIME ZONE'::REGTYPE;                                                              
SELECT pg_typeof(anon.random_date_between('01/01/0001'::DATE,'01/01/4001'::DATE)) = 'TIMESTAMP WITH TIME ZONE'::REGTYPE;                                                                     
SELECT pg_typeof(anon.random_date()) = 'TIMESTAMP WITH TIME ZONE'::REGTYPE;  

-- Integer
SELECT pg_typeof(anon.random_int_between(1,3)) = 'INTEGER'::REGTYPE;

--
-- Personal Data (First Name, etc.)
--

-- First Name
SELECT pg_typeof(anon.random_first_name()) = 'TEXT'::REGTYPE;

-- Last Name                                                                                                                                          
SELECT pg_typeof(anon.random_last_name()) = 'TEXT'::REGTYPE;


-- Email
SELECT pg_typeof(anon.random_email()) = 'TEXT'::REGTYPE;

-- Phone
SELECT pg_typeof(anon.random_phone('0033')) = 'TEXT'::REGTYPE;
SELECT anon.random_phone(NULL) IS NULL;
SELECT pg_typeof(anon.random_phone()) = 'TEXT'::REGTYPE;

-- Location
SELECT pg_typeof(anon.random_city_in_country('France')) = 'TEXT'::REGTYPE;
SELECT anon.random_city_in_country('dfndjndjnjdnvjdnjvndjnvjdnvjdnjnvdnvjdnvj') IS NULL;
SELECT anon.random_city_in_country(NULL) IS NULL;                                                                                   
SELECT pg_typeof(anon.random_city()) = 'TEXT'::REGTYPE;                                                                                                               
SELECT pg_typeof(anon.random_region_in_country('Italy')) = 'TEXT'::REGTYPE;
SELECT anon.random_region_in_country('c,dksv,kdfsdnfvsjdnfjsdnjfndj') IS NULL;
SELECT anon.random_region_in_country(NULL) IS NULL;
SELECT pg_typeof(anon.random_region()) = 'TEXT'::REGTYPE;                                                                                 
SELECT pg_typeof(anon.random_country()) = 'TEXT'::REGTYPE; 


--
-- Company
--

SELECT pg_typeof(anon.random_company()) = 'TEXT'::REGTYPE;

--
-- IBAN
--
SELECT pg_typeof(anon.random_iban()) = 'TEXT'::REGTYPE;

--
-- End
--
DROP EXTENSION anon;
