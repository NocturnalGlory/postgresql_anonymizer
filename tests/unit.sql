CREATE EXTENSION IF NOT EXISTS anon;

-- zip
SELECT pg_typeof(anon.random_zip());


--
-- string
--

SELECT pg_typeof(anon.random_string(1));
--SELECT anon_string(123456789);

--
-- First Name
--
SELECT pg_typeof(anon.random_first_name());

--
-- Company
--
SELECT pg_typeof(anon.random_company());

--
-- Date
--
SELECT pg_typeof(anon.random_date_between('01/01/1900'::TIMESTAMP WITH TIME ZONE,now()));
SELECT pg_typeof(anon.random_date_between('01/01/0000'::DATE,'O1/O1/4000'::DATE);
SELECT pg_typeof(anon.random_date());
