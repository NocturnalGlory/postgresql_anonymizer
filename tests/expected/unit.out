CREATE EXTENSION IF NOT EXISTS tsm_system_rows;
CREATE EXTENSION IF NOT EXISTS anon;
--                                                                                                                                                     
-- Generic Types                                                                                                                                       
--     
-- zip
SELECT pg_typeof(anon.random_zip()) = 'TEXT'::REGTYPE;
 ?column? 
----------
 t
(1 row)

-- string
SELECT pg_typeof(anon.random_string(1)) = 'TEXT'::REGTYPE;
 ?column? 
----------
 t
(1 row)

--SELECT anon_string(123456789);
-- Date                                                                                                                                                
SELECT pg_typeof(anon.random_date_between('01/01/1900'::TIMESTAMP WITH TIME ZONE,now())) = 'TIMESTAMP WITH TIME ZONE'::REGTYPE;                                                              
 ?column? 
----------
 t
(1 row)

SELECT pg_typeof(anon.random_date_between('01/01/0001'::DATE,'01/01/4001'::DATE)) = 'TIMESTAMP WITH TIME ZONE'::REGTYPE;                                                                     
 ?column? 
----------
 t
(1 row)

SELECT pg_typeof(anon.random_date()) = 'TIMESTAMP WITH TIME ZONE'::REGTYPE;  
 ?column? 
----------
 t
(1 row)

-- Integer
SELECT pg_typeof(anon.random_int_between(1,3)) = 'INTEGER'::REGTYPE;
 ?column? 
----------
 t
(1 row)

--
-- Personal Data (First Name, etc.)
--
-- First Name
SELECT pg_typeof(anon.random_first_name()) = 'TEXT'::REGTYPE;
 ?column? 
----------
 t
(1 row)

-- Last Name                                                                                                                                          
SELECT pg_typeof(anon.random_last_name()) = 'TEXT'::REGTYPE;
 ?column? 
----------
 t
(1 row)

-- Email
SELECT pg_typeof(anon.random_email()) = 'TEXT'::REGTYPE;
 ?column? 
----------
 t
(1 row)

-- Phone
SELECT pg_typeof(anon.random_phone('0033')) = 'TEXT'::REGTYPE;
 ?column? 
----------
 t
(1 row)

SELECT anon.random_phone(NULL) IS NULL;
 ?column? 
----------
 t
(1 row)

SELECT pg_typeof(anon.random_phone()) = 'TEXT'::REGTYPE;
 ?column? 
----------
 t
(1 row)

-- Location
SELECT pg_typeof(anon.random_city_in_country('France')) = 'TEXT'::REGTYPE;
 ?column? 
----------
 t
(1 row)

SELECT anon.random_city_in_country('dfndjndjnjdnvjdnjvndjnvjdnvjdnjnvdnvjdnvj') IS NULL;
 ?column? 
----------
 t
(1 row)

SELECT anon.random_city_in_country(NULL) IS NULL;                                                                                   
 ?column? 
----------
 t
(1 row)

SELECT pg_typeof(anon.random_city()) = 'TEXT'::REGTYPE;                                                                                                               
 ?column? 
----------
 t
(1 row)

SELECT pg_typeof(anon.random_region_in_country('Italy')) = 'TEXT'::REGTYPE;
 ?column? 
----------
 t
(1 row)

SELECT anon.random_region_in_country('c,dksv,kdfsdnfvsjdnfjsdnjfndj') IS NULL;
 ?column? 
----------
 t
(1 row)

SELECT anon.random_region_in_country(NULL) IS NULL;
 ?column? 
----------
 t
(1 row)

SELECT pg_typeof(anon.random_region()) = 'TEXT'::REGTYPE;                                                                                 
 ?column? 
----------
 t
(1 row)

SELECT pg_typeof(anon.random_country()) = 'TEXT'::REGTYPE; 
 ?column? 
----------
 t
(1 row)

--
-- Company
--
SELECT pg_typeof(anon.random_company()) = 'TEXT'::REGTYPE;
 ?column? 
----------
 t
(1 row)

--
-- IBAN
--
SELECT pg_typeof(anon.random_iban()) = 'TEXT'::REGTYPE;
 ?column? 
----------
 t
(1 row)

--
-- SIRET
--
SELECT pg_typeof(anon.random_siret()) = 'TEXT'::REGTYPE;
 ?column? 
----------
 t
(1 row)

SELECT pg_typeof(anon.random_siren()) = 'TEXT'::REGTYPE;
 ?column? 
----------
 t
(1 row)

--
-- End
--
DROP EXTENSION anon;
